typescript/sdk/src/HyperlaneApp.ts

```diff
@@ -26,39 +26,39 @@ export class HyperlaneApp<
     super(connectedContractsMap);
   }
 
-  static buildContracts<C extends HyperlaneContracts>(
-    addresses: ChainMap<HyperlaneAddresses>,
-    factories: HyperlaneFactories,
+  static buildContracts<F extends HyperlaneFactories>(
+    addresses: HyperlaneAddressesMap<any>,
+    factories: F,
     multiProvider: MultiProvider,
-  ): { contracts: ChainMap<C>; intersectionProvider: MultiProvider } {
-    const chains = Object.keys(addresses);
+  ): {
+    contracts: HyperlaneContractsMap<F>;
+    intersectionProvider: MultiProvider;
```

Comment:
why even put this on the class

Reply:
b/c every subclass uses it when calling `fromEnvironment`